packaging/version.py

```diff
@@ -198,7 +198,12 @@ class Version(_BaseVersion):
 
     def __init__(self, version):
         # Validate the version and parse it into pieces
-        match = self._regex.search(version)
+        try:
+            match = self._regex.search(version)
+        except TypeError:
+            raise InvalidVersion(
+                "Invalid type: '{0}' is not a string".format(type(version))
```

Comment:
Giving the `version`'s `type` is nice but I think `version`'s `repr` should also be given to help the debugging.